code/API_definitions/qos-profiles.yaml

```
@@ -0,0 +1,475 @@
+openapi: 3.0.3
+info:
+  title: QoS Profiles
+  description: |
+    The QoS Profiles API allows the discovery of Quality of Service (QoS) profiles and its characteristics.
+
+    # Introduction
+
+    QoS Profiles are used in conjunction with other Quality-On-Demand APIs, to offer the application developers the capability to request for stable latency (reduced jitter) or throughput for some specified application data flows between application clients (within a user device) and Application Servers (backend services). The developer has a pre-defined set of Quality of Service (QoS) profiles which they could choose from depending on their latency or throughput requirements. The QoS profiles are defined by the network operator and are mapped to the connectivity characteristics of the network.
+
+    # API functionality
+
+    The QoS Profiles API provides the following functionality:
+    - Discovery of all QoS profiles offered by the network operator
+    - Retrieve the characteristics of a specific QoS profile by name
+
+    How QoS profiles are mapped to connectivity characteristics are subject to agreements between the communication service provider and the API invoker. Within the CAMARA project, you can find a sample for such a mapping of QoS profiles. [CAMARA QoS Profiles Mapping Table (REFERENCE DRAFT)](https://github.com/camaraproject/QualityOnDemand/blob/main/documentation/API_documentation/QoSProfile_Mapping_Table.md)
+
+    # Further info and support
+
+    (FAQs will be added in a later version of the documentation)
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: project-email@sample.com
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: wip
+externalDocs:
+  description: Product documentation at Camara
+  url: https://github.com/camaraproject/
+security:
+  - oAuth2ClientCredentials: []
+servers:
+  - url: "{apiRoot}/qos-profiles/v-wip"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root, defined by the service provider, e.g. `api.example.com` or `api.example.com/somepath`
+tags:
+  - name: QoS Profiles
+    description: Manage QoS Profiles
```

```suggestion
    description: Query available QoS Profiles
```

We're only supporting get, post or delete calls that would modify the profile.  This is probably a holdover that was missed when we reduced the scope of the initial API.